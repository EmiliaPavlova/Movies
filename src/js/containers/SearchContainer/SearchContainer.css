.label {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 220px;
  text-transform: uppercase;
  color: #FFF;
  font-weight: bold;
  font-size: 12px;
}

.searchWrapper {
  width: 100%;
  margin-bottom: 15px;
}

.condition {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 26px;
}

.searchButton {
  padding: 3px 15px;
  border: none;
  text-transform: uppercase;
  color: #FFF;
  background: #3C3C3C;
}

.searchButton:active {
  background: #AD3A46;
}

.searchButton:focus {
  background: #F55263;
}

.submitButton {
  padding: 5px 50px;
  font-size: 14px;
  color: #FFF;
  background: #F55263;
}