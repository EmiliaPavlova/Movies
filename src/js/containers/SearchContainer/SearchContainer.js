import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Search from '../../components/Search/Search';
import Button from '../../components/Button/Button';
import './SearchContainer.css';

const conditions = ['Title', 'Genre'];

class SearchContainer extends Component {

  render() {
    const actionButton = <Button title='Search' type='submit' className='submitButton' onClick={() => console.log('click Search')} />;

    const renderConditions = conditions => {
      return (
        <div className='label'>
          Search by
          {conditions.map((item, index) => {
            return (
              <div key={index} className='condition'>
                {/* TODO: add logic for selected */}
                <Button title={item} className='searchButton' onClick={() => console.log(`click ${item}`)} />
              </div>
            )
          })}
        </div>
      )
    }

    return (
      <div className='searchWrapper'>
        <Search
          title='Find your movie'
          conditions={renderConditions(conditions)}
          actionButton={actionButton} />
      </div>
    )
  }
}

SearchContainer.propTypes = {
  // conditions
}

export default SearchContainer;
