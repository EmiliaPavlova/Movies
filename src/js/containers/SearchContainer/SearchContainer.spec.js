import React from 'react';
import { mount, shallow } from 'enzyme';
import SearchContainer from './SearchContainer';
import './SearchContainer.css';

const componentProps = (overrides = {}) => ({
  onClick: jest.fn(),
  ...overrides
});

const mountSearchContainer = props => mount(<SearchContainer {...props} />);

describe('SearchContainer', () => {
  describe('SearchContainer attributes', () => {
    it('matches the snapshot', () => {
      const wrapper = mountSearchContainer();
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Event handlers', () => {
    it('calls onClick on click event', () => {
      const onClickMock = jest.fn();
      const props = componentProps({ onClick: onClickMock });
      const wrapper = mountSearchContainer(props);
      const button = wrapper.find('button').at(0);
      button.simulate('click');
      // expect(onClickMock).toHaveBeenCalled();

      const actionButton = wrapper.find('button').at(2);
      actionButton.simulate('click');
      // expect(onClickMock).toHaveBeenCalled();
    });
  });
});
