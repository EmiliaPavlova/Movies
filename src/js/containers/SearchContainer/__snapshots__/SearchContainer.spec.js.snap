// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer SearchContainer attributes matches the snapshot 1`] = `
<SearchContainer>
  <div
    className="searchWrapper"
  >
    <Search
      actionButton={
        <Button
          className="submitButton"
          onClick={[Function]}
          title="Search"
          type="submit"
        />
      }
      conditions={
        <div
          className="label"
        >
          Search by
          <div
            className="condition"
          >
            <Button
              className="searchButton"
              onClick={[Function]}
              title="Title"
            />
          </div>
          <div
            className="condition"
          >
            <Button
              className="searchButton"
              onClick={[Function]}
              title="Genre"
            />
          </div>
        </div>
      }
      title="Find your movie"
    >
      <div
        className="title"
      >
        Find your movie
      </div>
      <input />
      <div
        className="buttons"
      >
        <div
          className="label"
        >
          Search by
          <div
            className="condition"
            key="0"
          >
            <Button
              className="searchButton"
              onClick={[Function]}
              title="Title"
            >
              <button
                className="button searchButton "
                onClick={[Function]}
              >
                Title
              </button>
            </Button>
          </div>
          <div
            className="condition"
            key="1"
          >
            <Button
              className="searchButton"
              onClick={[Function]}
              title="Genre"
            >
              <button
                className="button searchButton "
                onClick={[Function]}
              >
                Genre
              </button>
            </Button>
          </div>
        </div>
        <span
          className="actionButton"
        >
          <Button
            className="submitButton"
            onClick={[Function]}
            title="Search"
            type="submit"
          >
            <button
              className="button submitButton "
              onClick={[Function]}
              type="submit"
            >
              Search
            </button>
          </Button>
        </span>
      </div>
    </Search>
  </div>
</SearchContainer>
`;
